src/main/java/net/fabricmc/loader/transformer/accessWidener/AccessWidener.java

```diff
@@ -0,0 +1,355 @@
+		return methodAccess.getOrDefault(entryTriple, MethodAccess.DEFAULT);
+	}
+
+	public Set<String> getTargets() {
+		return classes;
+	}
+
+	private static int makePublic(int i) {
+		return (i & ~(Opcodes.ACC_PRIVATE | Opcodes.ACC_PROTECTED)) | Opcodes.ACC_PUBLIC;
+	}
+
+	private static int makeProtected(int i) {
+		if ((i & Opcodes.ACC_PUBLIC) != 0) {
+			//Return i if public
+			return i;
+		}
+
+		return (i & ~(Opcodes.ACC_PRIVATE)) | Opcodes.ACC_PROTECTED;
+	}
+
+	private static int makeFinalIfPrivate(int i) {
+		if ((i & Opcodes.ACC_PRIVATE) != 0) {
+			return i | Opcodes.ACC_FINAL;
+		}
+
+		return i;
+	}
+
+	private static int removeFinal(int i) {
+		return i & ~Opcodes.ACC_FINAL;
+	}
+
+	public interface Access {
+		Access makeAccessible();
+
+		Access makeExtendable();
+
+		Access makeMutable();
+
+		int apply(int i);
+	}
+
+	public enum ClassAccess implements Access {
+		DEFAULT(i -> i),
+		ACCESSIBLE(i -> makePublic(makeFinalIfPrivate(i))),
+		EXTENDABLE(i -> makePublic(removeFinal(i))),
+		ACCESSIBLE_EXTENDABLE(i -> makePublic(removeFinal(i)));
+
+		private final IntUnaryOperator operator;
+
+		ClassAccess(IntUnaryOperator operator) {
+			this.operator = operator;
+		}
+
+		@Override
+		public Access makeAccessible() {
+			if (this == EXTENDABLE || this == ACCESSIBLE_EXTENDABLE) {
+				return ACCESSIBLE_EXTENDABLE;
+			}
+
+			return ACCESSIBLE;
+		}
+
+		@Override
+		public Access makeExtendable() {
+			if (this == ACCESSIBLE || this == ACCESSIBLE_EXTENDABLE) {
+				return ACCESSIBLE_EXTENDABLE;
+			}
+
+			return EXTENDABLE;
+		}
+
+		@Override
+		public Access makeMutable() {
+			throw new UnsupportedOperationException("Classes cannot be made mutable");
+		}
+
+		@Override
+		public int apply(int i) {
+			return operator.applyAsInt(i);
+		}
+	}
+
+	public enum MethodAccess implements Access {
+		DEFAULT(i -> i),
+		ACCESSIBLE(i -> makePublic(makeFinalIfPrivate(i))), //Make public, add final if private
+		EXTENDABLE(i -> makeProtected(removeFinal(i))), //Make protected and strip final
+		ACCESSIBLE_EXTENDABLE(i -> makePublic(removeFinal(i))); //Make public and strip final
+
+		private final IntUnaryOperator operator;
+
+		MethodAccess(IntUnaryOperator operator) {
+			this.operator = operator;
+		}
+
+		@Override
+		public Access makeAccessible() {
+			if (this == EXTENDABLE || this == ACCESSIBLE_EXTENDABLE) {
+				return ACCESSIBLE_EXTENDABLE;
+			}
+
+			return ACCESSIBLE;
+		}
+
+		@Override
+		public Access makeExtendable() {
+			if (this == ACCESSIBLE || this == ACCESSIBLE_EXTENDABLE) {
+				return ACCESSIBLE_EXTENDABLE;
+			}
+
+			return EXTENDABLE;
+		}
+
+		@Override
+		public Access makeMutable() {
+			throw new UnsupportedOperationException("Methods cannot be made mutable");
+		}
+
+		@Override
+		public int apply(int i) {
+			return operator.applyAsInt(i);
+		}
+	}
+
+	public enum FieldAccess implements Access {
+		DEFAULT(i -> i),
+		ACCESSIBLE(i -> makePublic(i)),
```

Comment:
What if I want to make a field accessible as a subclass (protected) but not exposed to other users?

Reply:
`extendable` doesnt really fit for fields, so I think this is fine.